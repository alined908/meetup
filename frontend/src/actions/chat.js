import * as types from "../constants/action-types"
import {axiosClient} from '../accounts/axiosClient'

export const getRoom = (uri) => async dispatch => {
    dispatch({type: types.GET_ROOMS_REQUEST})
    try {
        const response = await axiosClient.get(
            `/api/chats/${uri}/`, {headers: {
                "Authorization": `Bearer ${localStorage.getItem('token')}`
            }}
        )
        dispatch({type: types.GET_ROOMS_SUCCESS, payload: response.data})
    } catch(e){
        dispatch({type: types.GET_ROOMS_ERROR, payload: "Unable to get chat room."})
    }
}   

export const getRooms = () => async dispatch => {
    dispatch({type: types.GET_ROOMS_REQUEST})
    try {
        const response = await axiosClient.get(
            "/api/chats/", {headers: {
                "Authorization": `Bearer ${localStorage.getItem('token')}`
            }}
        )
        dispatch({type: types.GET_ROOMS_SUCCESS, payload: response.data})
    } catch(e){
        dispatch({type: types.GET_ROOMS_ERROR, payload: "Unable to get chat rooms."})
    }
}

export const getMessages = (room) => async dispatch => {
    dispatch({type: types.GET_MESSAGES_REQUEST})
    try {
        const response = await axiosClient.get(
            `/api/chats/${room}/messages/`, {headers: {
            "Authorization": `Bearer ${localStorage.getItem('token')}`
        }})
        dispatch({type: types.GET_MESSAGES_SUCCESS, payload: response.data})
    } catch(e){
        dispatch({type: types.GET_MESSAGES_ERROR, payload: "Unable to get messages."})
    }
}

export const getMoreMessages = (room, last_message_id) => async dispatch => {
    dispatch({type: types.GET_MORE_MESSAGES_REQUEST})
    try {
        const response = await axiosClient.get(
            `/api/chats/${room}/messages/`, {params: {last: last_message_id}, headers: {
                "Authorization": `Bearer ${localStorage.getItem('token')}`
        }})
        dispatch({type: types.GET_MORE_MESSAGES_SUCCESS, payload: response.data})
    } catch(e){
        dispatch({type: types.GET_MORE_MESSAGES_ERROR, payload: "Unable to get more messages."})
    }

}

export const setActiveRoom = (room) => {
    return {
        type: types.SET_ACTIVE_ROOM,
        payload: {"uri": room}
    }
}

export const removeActiveRoom = () => {
    return {type: types.REMOVE_ACTIVE_ROOM}
}

export const updateRoom = (event) => async dispatch => {
    dispatch({type: types.UPDATE_ROOM, payload: event.message})
}

export const setTypingValue = (value) => {
    return {
        type: types.SET_TYPING_VALUE,
        payload: value
    }
}

export const addMessage = (message) => async dispatch => {
    dispatch({type: types.ADD_MESSAGE, payload: {message}})    
}