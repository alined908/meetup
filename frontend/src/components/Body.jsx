import React, {Component} from 'react';
import {Route, Switch} from 'react-router-dom';
import LogoutComponent from "./auth/LogoutComponent";
import ErrorComponent from "./auth/ErrorComponent";
import RegisterComponent from "./auth/RegisterComponent";
import WelcomeComponent from "./WelcomeComponent";
import ChatComponent from "./chat/ChatComponent"
import MeetupsComponent from "./meetup/MeetupsComponent";
import HomeComponent from "./HomeComponent";
import MeetupPageComponent from "./meetup/MeetupPageComponent"
import FriendsComponent from './invite/FriendsComponent';
import MeetupEventForm from "./meetup/MeetupEventForm"
import AuthenticatedRoute from "../accounts/AuthenticatedRoute";
import LoginComponent from "./auth/LoginComponent";
import Invites from "./invite/Invites"
import MeetupForm from "./meetup/MeetupForm"
import Profile from "./Profile"

class Body extends Component {

    render () {
        return (
            <div className="c">
                <Switch>
                    <Route path="/" exact component={HomeComponent}></Route>
                    <Route path="/login" component={LoginComponent}></Route>
                    <Route path="/register" component={RegisterComponent}></Route>
                    <AuthenticatedRoute path="/logout" component={LogoutComponent}></AuthenticatedRoute>
                    <AuthenticatedRoute path="/welcome" component={WelcomeComponent}></AuthenticatedRoute>
                    <AuthenticatedRoute path="/meetups/:uri/new" exact component={MeetupEventForm}></AuthenticatedRoute>
                    <AuthenticatedRoute path="/meetups/new" exact component={MeetupForm}></AuthenticatedRoute>
                    <AuthenticatedRoute path="/meetups/:uri" component={MeetupPageComponent}></AuthenticatedRoute>
                    <AuthenticatedRoute path="/meetups" exact component={MeetupsComponent}></AuthenticatedRoute>
                    <AuthenticatedRoute path="/chat/:uri" component={ChatComponent}></AuthenticatedRoute>
                    <AuthenticatedRoute path="/chat" exact component={ChatComponent}></AuthenticatedRoute>
                    <AuthenticatedRoute path="/friends" component={FriendsComponent}></AuthenticatedRoute>
                    <AuthenticatedRoute path="/invites" component={Invites}></AuthenticatedRoute>
                    <AuthenticatedRoute path="/profile/:id" component={Profile}></AuthenticatedRoute>
                    <Route component={ErrorComponent}></Route>
                </Switch>
            </div>
        )
    }
}

export default Body