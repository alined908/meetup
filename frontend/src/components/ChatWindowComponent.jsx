import React, {Component} from 'react'
import ChatMessageComponent from "./ChatMessageComponent"
import ChatInputComponent from "./ChatInputComponent"
import {connect} from 'react-redux'

class ChatWindowComponent extends Component {

    componentDidMount() {
        console.log("mounted chat window")
        this.scrollToBottom();
      }
    componentDidUpdate() {
        console.log("updated chat window")
        this.scrollToBottom();
    }
    scrollToBottom = () => {
        console.log("scroll to bottom called")
        this.messagesEnd && this.messagesEnd.scrollIntoView({ behavior: "smooth" });
    };
    
    render () {
        return (
            <div className="chat-window" ref={this.chatsRef}>
                <div className="chat-header">
                    Conversation for uri: {this.props.activeRoom}
                </div>
                <div className="chat-messages">
                    {this.props.messages && this.props.messages.map((msg) => <ChatMessageComponent user={this.props.user} message={msg.message} members={this.props.activeChatMembers}/>)}
                </div>
                {this.props.activeChatMembers && <div ref={(el) => { this.messagesEnd = el; }} className="chat-input">
                    <ChatInputComponent/>
                </div>}
            </div>
        )
    }
}

function mapStateToProps(state){
    if (state.chat.activeRoom in state.chat.rooms){
        return {
            activeChatMembers: state.chat.rooms[state.chat.activeRoom].members,
            user: state.auth.user
        }
    } else {
        return {
            activeChatMembers: null,
            user: state.auth.user
        }
    }
    
}


export default connect(mapStateToProps)(ChatWindowComponent)