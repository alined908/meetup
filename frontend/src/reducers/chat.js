import {GET_ROOMS_REQUEST, GET_ROOMS_SUCCESS, GET_ROOMS_ERROR, 
    GET_MESSAGES_REQUEST, GET_MESSAGES_SUCCESS, GET_MESSAGES_ERROR, 
    GET_MORE_MESSAGES_REQUEST, GET_MORE_MESSAGES_SUCCESS, GET_MORE_MESSAGES_ERROR,
    CLEAR_STORE, REMOVE_ACTIVE_ROOM, ADD_ROOM, SET_ACTIVE_ROOM, SET_TYPING_VALUE, ADD_MESSAGE} from "../constants/action-types"

const defaultState = {
    rooms: {},
    messages: [],
    activeRoom: null,
    setTypingValue: "",
    isActiveRoomSet: false,
    isRoomsInitialized: false,
    isRoomsFetching: false,
    isMessagesInitialized: false,
    isMessagesFetching: false,
    isMoreMessagesFetching: false
}

export default function chatReducer(state = defaultState, action){
    switch(action.type){
        case GET_ROOMS_REQUEST:
            return {...state, isRoomsFetching: true}
        case GET_ROOMS_SUCCESS:
            return {...state, rooms: action.payload, isRoomsInitialized: true, isRoomsFetching: false}
        case GET_ROOMS_ERROR:
            return {...state, isRoomsFetching: false, errorMessage: action.payload.message}
        case GET_MESSAGES_REQUEST:
            return {...state, isMessagesFetching: true}
        case GET_MESSAGES_SUCCESS:
            return {...state, messages: action.payload, isMessagesInitialized: true, isMessagesFetching: false}
        case GET_MESSAGES_ERROR:
            return {...state, isMessagesFetching: false, errorMessage: action.payload.message}
        case GET_MORE_MESSAGES_REQUEST:
            return {...state, isMoreMessagesFetching: true}
        case GET_MORE_MESSAGES_SUCCESS:
            return {...state, messages: [...action.payload, ...state.messages], isMoreMessagesFetching: false}
        case GET_MORE_MESSAGES_ERROR:
            return {...state, errorMessage: action.payload.message, isMoreMessagesFetching: false}
        case ADD_ROOM:
            return {...state, rooms: state.rooms.concat([action.payload])}
        case SET_ACTIVE_ROOM:
            return {...state, activeRoom: action.payload.uri, isActiveRoomSet: true, isMessagesInitialized: false}
        case REMOVE_ACTIVE_ROOM:
            return {...state, activeRoom: null, isActiveRoomSet:false, isMessagesInitialized:false}
        case SET_TYPING_VALUE:
            return {...state, setTypingValue: action.payload}
        case ADD_MESSAGE:
            return {...state, setTypingValue: '', messages: [...state.messages, action.payload.message]}
        case CLEAR_STORE:
            return defaultState
        default:
            return state
    }
}