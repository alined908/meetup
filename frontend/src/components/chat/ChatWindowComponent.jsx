import React, {Component} from 'react'
import ChatMessageComponent from "./ChatMessageComponent"
import {connect} from 'react-redux'
import WebSocketInstance from '../../accounts/WebSocket'
import {setTypingValue, addMessage, getMessages} from "../../actions/chat";
import {Button} from '@material-ui/core'
import {Link} from 'react-router-dom';

class ChatWindowComponent extends Component {
    constructor(props){
        super(props)
        let user = this.props.user
        WebSocketInstance.addChatCallbacks(this.props.getMessages, this.props.addMessage)
        // WebSocketInstance.fetchMessages(this.props.activeRoom);
    }
    
    handleChange = (e) => {
        this.props.setTypingValue(e.target.value);        
    }

    handleSubmit = (e) => {
        if (e.key === "Enter"){
            e.preventDefault();
            console.log("handle submit")
            const messageObject = {from: this.props.user.id, text: this.props.message, room: this.props.room}
            console.log(messageObject)
            WebSocketInstance.newChatMessage(messageObject)
        }
    }

    componentDidMount() {
        this.scrollToBottom();
      }
    componentDidUpdate() {
        this.scrollToBottom();
    }
    scrollToBottom = () => {    
        this.messagesEnd && this.messagesEnd.scrollIntoView({ behavior: "smooth" });
    };
    
    render () {
        return (
            <div className="chat-window" ref={this.chatsRef}>
                <div className="chat-header">
                    {this.props.activeRoom && <Link to={`/meetups/${this.props.activeRoom}`}><Button color="primary">Go to Meetup</Button></Link>}
                </div>
                <div className="chat-messages">
                    {this.props.activeChatMembers && this.props.messages && this.props.messages.map((msg) => <ChatMessageComponent user={this.props.user} message={msg.message} members={this.props.activeChatMembers}/>)}
                </div>
                {this.props.activeChatMembers && <div ref={(el) => { this.messagesEnd = el; }} className="chat-input">
                    <form className="chat-input-form">
                        <input className="chat-input"
                            type="text"
                            onChange={this.handleChange} 
                            onKeyPress={this.handleSubmit}
                            value={this.props.message} 
                            placeholder="Type here to send a message">
                        </input>
                    </form>
                </div>}
            </div>
        )
    }
}

function mapStateToProps(state){
    if (state.chat.activeRoom in state.chat.rooms){
        return {
            activeChatMembers: state.chat.rooms[state.chat.activeRoom].members,
            user: state.user.user,
            message: state.chat.setTypingValue,
            room: state.chat.activeRoom,
        }
    } else {
        return {
            activeChatMembers: null,
            user: state.user.user,
        }
    }
    
}

const mapDispatchToProps = {
    setTypingValue,
    addMessage,
    getMessages
}


export default connect(mapStateToProps, mapDispatchToProps)(ChatWindowComponent)