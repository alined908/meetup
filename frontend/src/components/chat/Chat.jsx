import React, { Component } from 'react'
import { ChatBar, ChatWindow } from "../components"
import { connect } from "react-redux";
import {getMessages, setActiveRoom, addMessage, removeActiveRoom, getRooms} from "../../actions"
import WebSocketService from "../../accounts/WebSocket"
import AuthenticationService from '../../accounts/AuthenticationService';
import { Helmet } from "react-helmet"
import styles from '../../styles/chat.module.css'

class Chat extends Component {
    constructor(props){
        super(props)
        this.state = {
            socket: new WebSocketService(),
            nonMobile: window.matchMedia("(min-width: 768px)").matches
        }
        this.state.socket.addChatCallbacks(this.props.getMessages, this.props.addMessage)
    }

    componentDidMount(){
        const handler = e => this.setState({nonMobile: e.matches});
        window.matchMedia("(min-width: 768px)").addListener(handler);
        this.props.getRooms()
        if("uri" in this.props.match.params){
            this.getRelevantInfo(this.props.match.params.uri)
        }
    }

    componentDidUpdate(prevProps){
        if (this.props.match.params.uri !== prevProps.match.params.uri){
            if (this.state.socket.exists()){
                this.state.socket.disconnect()
            }
            this.getRelevantInfo(this.props.match.params.uri)
        }
    }

    componentWillUnmount(){
        this.props.removeActiveRoom()
        this.state.socket.disconnect()
    }

    getRelevantInfo(uri) {
        const socket = this.state.socket
        const token = AuthenticationService.retrieveToken()
        const path = `/ws/chat/${this.props.match.params.uri}/`;
        socket.connect(path, token);
        this.setRoomInfo(uri)
    }

    setRoomInfo(uri){
        this.props.setActiveRoom(uri);
        this.props.getMessages(uri);
    }

    render(){
        const renderChatWindow = () => {
            if (this.props.isMessagesInitialized || this.props.isMessagesFetching) {
                return <ChatWindow
                            socket={this.state.socket} isMessagesInitialized={this.props.isMessagesInitialized} 
                            activeRoom={this.props.activeRoom} messages={this.props.messages}
                        />
            } else {
                return <ChatWindow socket={this.state.socket} activeRoom={null} messages={[]}/>
            }
        }

        return (
            <div className={`${styles.chat}` + (this.state.nonMobile ? "" : "chat-mobile")}>
                <Helmet>
                    <meta charSet="utf-8" />
                    <title>Chat</title>
                    <meta name="description" content="Phoodie Chat" />
                </Helmet>
                <ChatBar rooms={this.props.rooms}/>
                {renderChatWindow()}
            </div>
        )
    }
}

const mapDispatchToProps = {
    setActiveRoom,
    getMessages,
    getRooms,
    addMessage,
    removeActiveRoom
}

function mapStateToProps(state){
    return {
        isMessagesInitialized: state.chat.isMessagesInitialized,
        activeRoom: state.chat.activeRoom,
        messages: state.chat.messages,
        isActiveRoomSet: state.chat.isActiveRoomSet,
        rooms: Object.values(state.chat.rooms),
        isMessagesFetching: state.chat.isMessagesFetching
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(Chat)
export {Chat as UnderlyingChat}