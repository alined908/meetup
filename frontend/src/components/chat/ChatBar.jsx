import React, {Component} from 'react';
import {Contact} from "../components";
import {connect} from 'react-redux';
import {IconButton, Tooltip, CircularProgress} from '@material-ui/core'
import {Event as EventIcon, Person as PersonIcon, Search as SearchIcon, Error as ErrorIcon} from '@material-ui/icons';
import PropTypes from 'prop-types';
import {chatRoomPropType, userPropType,} from "../../constants/prop-types"
import WebSocketService from "../../accounts/WebSocket"
import AuthenticationService from '../../accounts/AuthenticationService';
import {updateRoom} from '../../actions'
import styles from '../../styles/chat.module.css'

class ChatBar extends Component {
    constructor(props){
        super(props)
        this.state = {
            rooms: [],
            filteredRooms: [],
            searchedRooms: [],
            liveRooms: [],
            filters: {
                "friend": false, 
                "meetup": false
            },
            isFiltered: false,
            isSearched: false,
            searchInput: '',
            socket: new WebSocketService()
        }
        this.state.socket.addContactsCallbacks(this.props.updateRoom)
    }

    // Initialize socket for chat rooms.
    componentDidMount(){
        const socket = this.state.socket
        const token = AuthenticationService.retrieveToken()
        const path = `/ws/chat/rooms/`;
        socket.connect(path, token);
    }

    componentDidUpdate(prevProps) {
        if (this.props.rooms !== prevProps.rooms) {
            this.setState({rooms: this.props.rooms, liveRooms: this.props.rooms})
        }
    }

    filterAndSearch = () => {
        const rooms = this.state.rooms
        const filter = this.state.searchInput
        const filterMeetup = this.state.filters["meetup"]
        const filterFriend = this.state.filters["friend"]
        let newRooms;

        newRooms = rooms.filter((room) => {
        
            let meetupCriteria = false;
            if (room.meetup !== null) {
                meetupCriteria = room.name.toLowerCase().includes(filter.toLowerCase()) && !filterMeetup
            }

            let friendCriteria = false;
            if (room.friendship !== null){
                const user = this.props.user
                let friend;
                for (let key in room.members){
                    if (key !== user.id.toString()){
                        friend = room.members[key]
                    }
                }
                const friendName = `${friend.first_name} ${friend.last_name}`

                friendCriteria = (
                    (friendName.toLowerCase().includes(filter.toLowerCase())) || 
                    (friend.email.toLowerCase().includes(filter.toLowerCase()))
                ) && !filterFriend
            }

            return meetupCriteria || friendCriteria
        })

        this.setState({liveRooms: newRooms})
    }

    filterRooms = (type) => {
        let filters = {...this.state.filters}
        filters[type] = !filters[type]
        this.setState({filters}, () => this.filterAndSearch())
    }

    handleSearchInputChange = (e) => {
        var filter = e.target.value;
        this.setState({searchInput: filter}, () => this.filterAndSearch())
    }

    render (){
        const rooms = this.state.liveRooms

        return (
            <div className={styles.bar}>
                <div className={styles.barTop}>
                    <div>Contacts</div>
                    <div>
                        <Tooltip title="Friends">
                            <IconButton 
                                color={this.state.filters["friend"] ? "default" : "primary"} 
                                onClick={() => this.filterRooms("friend")} edge="end"
                            >
                                <PersonIcon/>
                            </IconButton>
                        </Tooltip>
                        <Tooltip title="Meetups">
                            <IconButton 
                                color={this.state.filters["meetup"] ? "default" : "primary"} 
                                onClick={() => this.filterRooms("meetup")}
                            >
                                <EventIcon/>
                            </IconButton>
                        </Tooltip>
                    </div>
                </div>
                <div className={styles.contacts}>
                    {this.props.isRoomsFetching && 
                        <div className="loading">
                            <CircularProgress/>
                        </div>
                    }
                    {(this.props.isRoomsInitialized && rooms.length === 0) && 
                        <div className="no-entity">
                            <ErrorIcon style={{color: "rgb(255, 212, 96)"}}/>
                            <span className="no-entity-text">
                                No chat rooms match criteria!
                            </span>
                        </div>
                    }
                    {this.props.isRoomsInitialized && rooms.map((room) => 
                        <Contact 
                            key={room.id}
                            room={room}
                            user={this.props.user} 
                            currentRoom={this.props.currentRoom}
                        />
                    )}
                </div>
                <div className={styles.search}>
                    <SearchIcon/>
                    <input 
                        type="text"
                        className={styles.input} 
                        placeholder="Search Contacts..." 
                        value={this.state.searchInput} 
                        onChange={this.handleSearchInputChange}
                    />
                </div>
            </div>
        )
    }
}

ChatBar.propTypes = {
    rooms: PropTypes.arrayOf(
        chatRoomPropType
    ),
    user: userPropType,
    isRoomsInitialized: PropTypes.bool
}

const mapStateToProps = (state) => {
    return {
        user: state.user.user,
        isRoomsInitialized: state.chat.isRoomsInitialized,
        isRoomsFetching: state.chat.isRoomsFetching,
        currentRoom: state.chat.activeRoom
    }
}

const mapDispatchToProps = {
    updateRoom
}

export default connect(mapStateToProps, mapDispatchToProps)(ChatBar)
export {ChatBar as UnderlyingChatBar}