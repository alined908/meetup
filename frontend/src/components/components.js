//Auth
import ErrorComponent from "./auth/ErrorComponent";
import LoginComponent from "./auth/LoginComponent";
import LogoutComponent from "./auth/LogoutComponent";
import RegisterComponent from "./auth/RegisterComponent";

//Chat
import ChatBarComponent from "./chat/ChatBarComponent"
import ChatComponent from "./chat/ChatComponent"
import ChatMessageComponent from "./chat/ChatMessageComponent"
import ChatWindowComponent from "./chat/ChatWindowComponent"
import ContactComponent from "./chat/ContactComponent"

//invite
import Friend from './invite/Friend';
import FriendsComponent from './invite/FriendsComponent';
import Invite from "./invite/Invite"
import Invites from "./invite/Invites"

//meetup
import Map from "./meetup/Map"
import Meetup from "./meetup/Meetup";
import MeetupCard from "./meetup/MeetupCard";
import MeetupEvent from "./meetup/MeetupEvent";
import MeetupEventForm from "./meetup/MeetupEventForm"
import MeetupForm from "./meetup/MeetupForm";
import MeetupFriend from "./meetup/MeetupFriend";
import MeetupPageComponent from "./meetup/MeetupPageComponent"
import MeetupsComponent from "./meetup/MeetupsComponent";
import Restauraunt from "./meetup/Restauraunt";

//fields
import CategoryAutocomplete from "./fields/CategoryAutocomplete"
import Location from "./fields/Location"
import renderTextField from "./fields/renderTextField"
import renderDatePicker from "./fields/renderDatePicker"
import renderDateSimplePicker from "./fields/renderDateSimplePicker"
import UserAutocomplete from "./fields/UserAutocomplete"

//General
import Body from "./general/Body"
import GlobalMessage from "./general/GlobalMessage"
import HomeComponent from "./general/HomeComponent";
import LiveUpdatingBadge from "./general/LiveUpdatingBadge"
import Navigation from "./general/Navigation"
import Profile from "./general/Profile"
import GroupAvatars from './general/GroupAvatars'
import NotificationWrapper from "./general/NotificationWrapper"
import CalendarComponent from "./general/CalendarComponent"
import CategoriesComponent from "./general/CategoriesComponent"
import CategoryComponent from "./general/CategoryComponent"
import Preferences from "./general/Preferences"
import Preference from "./general/Preference"
import SettingsComponent from "./general/SettingsComponent"

export {
    ErrorComponent, LoginComponent, LogoutComponent, RegisterComponent, 
    ChatBarComponent, ChatComponent, ChatMessageComponent, ChatWindowComponent, ContactComponent,
    Friend, FriendsComponent, Invite, Invites,
    Map, Meetup, MeetupCard, MeetupEvent, MeetupEventForm, MeetupFriend, MeetupForm, MeetupPageComponent, MeetupsComponent, Restauraunt,
    CategoryAutocomplete, UserAutocomplete, Location, renderTextField, renderDatePicker, renderDateSimplePicker,
    Body, GlobalMessage, GroupAvatars, HomeComponent, LiveUpdatingBadge, Navigation, Profile, NotificationWrapper, 
    Preferences, Preference, CalendarComponent, CategoriesComponent, CategoryComponent, SettingsComponent}