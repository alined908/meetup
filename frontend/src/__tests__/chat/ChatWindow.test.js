import React from 'react'
import {shallow, mount} from 'enzyme';
import {ChatWindowComponent} from "../../components/components" 
import {UnderlyingChatWindowComponent} from "../../components/chat/ChatWindowComponent"

Date.now = jest.fn(() => new Date(Date.UTC(2017, 1, 14)).valueOf())
const user = {id: 1, email: "example@gmail.com", first_name: "Daniel", last_name: "Lee", avatar: null}
const user2 = {id: 2, email: "example2@gmail.com", first_name: "Bob", last_name: "Jim", avatar: null}
const members = {1: user,2: user2}
const rooms = {
    friend: {id: 1, uri: "abc", name: "Friend", timestamp: Date.now().toString(), members: members, friendship: 1, meetup:null, notifs: 0},
    meetup: {id: 2, uri: "xyz", name: "Meetup", timestamp: Date.now().toString(), members: members, friendship: null, meetup:1, notifs: 3}
}
const message = {id: 1, message: "hello", timestamp: Date.now().toString(), is_read: false, room_id: 1, sender_id: 1}
const message2 = {id: 1, message: "hello", timestamp: Date.now().toString(), is_read: false, room_id: 1, sender_id: 1}
const messages = [message, message2]


describe("ChatWindowComponent unit ", () => {

    it('renders without crashing given props', () => {
        const props = {
            activeRoom: rooms.friend.uri,
            isMessagesInitialized: true,
            messages: messages,
            room: rooms.friend,
            activeChatMembers: members,
            user: user,
            socket: {}
        }
        const wrapper = shallow(<UnderlyingChatWindowComponent {...props}/>)
        expect(wrapper).toMatchSnapshot()
    })

    it ("shows link to meetup if room is friend" , () => {
        const props = {
            activeRoom: rooms.friend.uri,
            isMessagesInitialized: true,
            messages: messages,
            room: rooms.friend,
            activeChatMembers: members,
            user: user,
            socket: {}
        }
        const wrapper = shallow(<UnderlyingChatWindowComponent {...props}/>)
        expect(wrapper.text()).toContain('Profile')
    })

    it ("shows link to user if room is meetup" , () => {
        const props = {
            activeRoom: rooms.meetup.uri,
            isMessagesInitialized: true,
            messages: [],
            room: rooms.meetup,
            activeChatMembers: members,
            user: user,
            socket: {}
        }
        const wrapper = shallow(<UnderlyingChatWindowComponent {...props}/>)
        expect(wrapper.text()).toContain('Meetup')
    })
})