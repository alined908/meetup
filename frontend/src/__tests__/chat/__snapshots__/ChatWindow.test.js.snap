// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatWindowComponent unit  renders without crashing given props 1`] = `
<div
  className="chat-window elevate"
>
  <div
    className="chat-header"
  >
    <Link
      to="/profile/2"
    >
      <WithStyles(ForwardRef(Tooltip))
        title="Go to Profile"
      >
        <WithStyles(ForwardRef(Button))
          color="primary"
          startIcon={<UNDEFINED />}
        >
          Profile
        </WithStyles(ForwardRef(Button))>
      </WithStyles(ForwardRef(Tooltip))>
    </Link>
  </div>
  <div
    className="chat-messages-wrapper"
    onScroll={[Function]}
  >
    <div
      className="chat-messages"
    >
      <ChatMessageComponent
        key="1"
        members={
          Object {
            "1": Object {
              "avatar": null,
              "email": "example@gmail.com",
              "first_name": "Daniel",
              "id": 1,
              "last_name": "Lee",
            },
            "2": Object {
              "avatar": null,
              "email": "example2@gmail.com",
              "first_name": "Bob",
              "id": 2,
              "last_name": "Jim",
            },
          }
        }
        message={
          Object {
            "id": 1,
            "is_read": false,
            "message": "hello",
            "room_id": 1,
            "sender_id": 1,
            "timestamp": "1487030400000",
          }
        }
        user={
          Object {
            "avatar": null,
            "email": "example@gmail.com",
            "first_name": "Daniel",
            "id": 1,
            "last_name": "Lee",
          }
        }
      />
      <ChatMessageComponent
        key="1"
        members={
          Object {
            "1": Object {
              "avatar": null,
              "email": "example@gmail.com",
              "first_name": "Daniel",
              "id": 1,
              "last_name": "Lee",
            },
            "2": Object {
              "avatar": null,
              "email": "example2@gmail.com",
              "first_name": "Bob",
              "id": 2,
              "last_name": "Jim",
            },
          }
        }
        message={
          Object {
            "id": 1,
            "is_read": false,
            "message": "hello",
            "room_id": 1,
            "sender_id": 1,
            "timestamp": "1487030400000",
          }
        }
        user={
          Object {
            "avatar": null,
            "email": "example@gmail.com",
            "first_name": "Daniel",
            "id": 1,
            "last_name": "Lee",
          }
        }
      />
      <div />
    </div>
  </div>
  <div>
    <form
      className="chat-input-form"
    >
      <input
        className="chat-input"
        onChange={[Function]}
        onKeyPress={[Function]}
        placeholder="Type a message here..."
        type="text"
        value=""
      />
    </form>
    <div>
      <WithStyles(ForwardRef(Button))
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#FFD460",
            "borderRadius": 15,
            "fontFamily": "Lato",
            "fontSize": 10,
            "fontWeight": "700",
          }
        }
      >
        Send
      </WithStyles(ForwardRef(Button))>
    </div>
  </div>
</div>
`;
